cofmpy/data_stream_handler/kafka_data_stream_handler.py

```diff
@@ -57,65 +57,50 @@ def __init__(self, topic, uri, group_id, variable, **kwargs):
         """
 
         # Configuration handling
-        interp_method, self.timeout = self._validate_config(kwargs)
-        self.topic = topic
-        server_url, port = uri.split(":")
-        self.group_id = group_id
-        self.var_name = variable
-
-        # Consumer isntantiation
-        self._start_consumer(server_url, port, group_id)
-        self._subscribed = False
+        positional = {
+            "topic": topic,
+            "uri": uri,
+            "group_id": group_id,
+            "variable": variable,
+        }
+        kwargs.update(positional)
+        self.config = KafkaHandlerConfig.from_dict(kwargs)
+        logger.debug(f"Parsed config for {self}: {vars(self.config)}")
 
-        # Other variables
-        self.interpolator = Interpolator(interp_method)
-        self.data = pd.DataFrame(columns=["t", self.var_name])
-        self.consumer_thread = None
-        self.running = False
+        # Data-related instances
+        self.interpolator = Interpolator(self.config.interpolation)
+        self.data = pd.DataFrame(columns=["t", self.config.variable])
```

Comment:
Just thinking for later. pd.Dataframe could be replaced by a buffer/queue in order to keep only a limited number of data.

The DataStreamHandler (mainly those for real time) are not aimed to store the whole history of input data, but just keeping the latest data to feed to the Master.